editor.planx.uk/src/@planx/components/Checklist/model.test.ts

```diff
@@ -194,23 +194,194 @@ describe("Editor validation", () => {
+                },
+              ],
+            },
+            {
+              title: "Repeated group name",
+              children: [
+                {
+                  id: "123",
+                  data: { text: "123" },
+                },
+              ],
+            },
+          ],
+        }),
+      ).rejects.toThrow("Groups must have unique titles");
+    });
+
     describe("options without data values", () => {
-      test.todo("checklists - unique labels must be used for each option");
-      test.todo("grouped checklists - labels must be unique within groups");
-      test.todo("grouped checklists - labels can be repeated across groups");
+      test("checklists - unique labels must be used for each option", async () => {
+        await expect(() =>
+          validationSchema.validate({
+            title: "Test",
+            options: [
+              {
+                id: "a",
+                data: { text: "Non-unique label" },
+              },
+              {
+                id: "b",
+                data: { text: "Non-unique label" },
+              },
+            ],
+          }),
+        ).rejects.toThrow("Options must have unique labels");
+      });
+
+      test("grouped checklists - labels must be unique within groups", async () => {
+        await expect(() =>
+          validationSchema.validate({
+            title: "Test",
+            groupedOptions: [
+              {
+                title: "Section 1",
+                children: [
+                  {
+                    id: "a",
+                    data: { text: "Non-unique label" },
+                  },
+                  {
+                    id: "b",
+                    data: { text: "Non-unique label" },
+                  },
+                ],
+              },
+            ],
+          }),
+        ).rejects.toThrow(
+          "Options within a single group must have unique labels",
+        );
+      });
+
+      test("grouped checklists - labels can be repeated across groups", async () => {
+        const result = validationSchema.validate({
+          title: "Test",
+          groupedOptions: [
+            {
+              title: "Section 1",
+              children: [
+                {
+                  id: "a",
+                  data: { text: "Non-unique label" },
+                },
+                {
+                  id: "b",
+                  data: { text: "Foo" },
+                },
+              ],
+            },
+            {
+              title: "Section 2",
+              children: [
+                {
+                  id: "a",
+                  data: { text: "Non-unique label" },
+                },
+                {
+                  id: "b",
+                  data: { text: "Bar" },
+                },
+              ],
+            },
+          ],
+        });
+
+        expect(result).toBeTruthy();
+      });
     });
 
     describe("options with data values", () => {
```

Comment:
nit: can we add another mock the realistic valid example from the Slack thread here which actually repeats `val` with unique labels? All these mocks also have unique data values, which is easy to misread a a requirement here IMO ! 

Eg 
```
options: [
    {
      id: "yes",
      data: { text: "Yes", val: "true" },
    },
    {
      id: "no",
      data: { text: "No", val: "false" },
    },
    {
      id: "unsure",
      data: { text: "I don't know", val: "false" },
    },
  ],
```

Reply:
Similarly, would like to see a mock that confirms label white-space is trimmed - eg ` text: "Option A"` and `text: "Option A "` should _not_ be unique!